### DISCUSSION TITLE
No Suricata Alerts from PCAP via tcpreplay

### DISCUSSION THREAD

**Discussion by boecks** [Security-Onion-Solutions/securityonion#2613] [Unsupported Versions]
Hello!

I have a fresh install of SecurityOnion 2.3 and try to get Suricata alerts when I use tcpreplay to replay the following PCAP: https://github.com/tatsui-geek/malware-traffic-analysis.net/blob/master/2018-01-08-Loki-Bot-traffic.pcap 

Unfortunately, I get no alerts, although, ther should be some. I have tested it with other setups using Suricata.

I also tried so-import-pcap without luck. I get some events from Zeek but none from Suricata.

The proposed solution from https://github.com/Security-Onion-Solutions/securityonion/discussions/2489#discussioncomment-252599 did not work either.

However, I can confirm, that my setup is generally working. (i.e. the monitoring interface is capturing the traffic etc.)

Any help would be highly appreciated. I need this to work in order to proceed with my masters thesis... ^^

Regards, Sascha

**Comment by weslambert** [comment]
Are you able to provide evidence of specific rules/signatures that should be firing (ex. from other setups)?  I'd ensure you are using the same ruleset, and that the rules you are expecting to generate alerts are actually enabled.

**Reply by boecks** [reply]
The rules in question are included in /opt/so/conf/suricata/rules/all.rules - I double checked that and compared the rules with those from https://rules.emergingthreats.net/open/suricata-5.0/emerging-all.rules

According to the following screenshot from /opt/so/saltstack/local/pillar/mionions/securityonion_eval.sls, there should be no rule explicity disabled. Or did I get the concept wrong here?

![idstools](https://user-images.githubusercontent.com/47642539/104578245-c78ec500-565a-11eb-83e6-f031d7ddeb89.png)

**Reply by boecks** [reply]
In addition to my previous answer, I want to add the specific rules, taken from the rules file of SO.

https://pastebin.com/nKgCCEs9

These rules are also present in the ruleset for RockNSM

To make things clearer, I also want to line out, what I am doing:

1. SO is running, verify alert with `curl http://testmyids.com` from my victim machine - get an alert
2. Also verify with `curl -sSL https://raw.githubusercontent.com/0xtf/testmynids.org/master/tmNIDS -o /tmp/tmNIDS && chmod +x /tmp/tmNIDS && /tmp/tmNIDS` from my victim machine - get several alerts using option 11 "Chaos"
2. tcpreplay -i enp0s20 2018-01-08-Loki-Bot-traffic.pcap
 - where enp0s20 is the physical interface of my victrim machine and the pcap is from my https://github.com/Security-Onion-Solutions/securityonion/discussions/2613#discussion-1764980 post, containing LokiBot traffic
 - replaying this, fires alerts in SELKS6 and RockNSM 2.5.1

I hope, I provided enough input now, to find out why this is not working in SO

Thank you!

**Comment by defensivedepth** [comment]
Also, what interface are you replaying to?

**Comment by boecks** [comment]
Thank you for your quick replys!

To answer your questions:

- I tested the same PCAP with Suricata in SELKS6 and RockNSM 2.5.1, both firing the alarms. All my setups are fresh off the shelfes, so no tuning from my side...
- I checked the rules and can verify, that all Setups used an up-to-date set of ETOPEN. I even did a deep comparison and checked if the rules are present in the all.rules file -> they are
- I have an Ubuntu machine in my test-lab, that is replaying to its physical interface. The traffic of this and other devices is mirrored to the monitoring-interface of my SO machine through port mirroring of my switch. This setup is working, since testmyids and other generic noise-alerts are firing in Suricata.


What else have I checked:
- I noticed Suricata has some different settings in suricata.yaml in SELKS6
- SELKS6 is one major release of Suricata ahead from SO. They have Suricata 6 running.

Thank you again for your help!

Best, Sascha

Edit: 

This is my lab setup. The network sensor is Security Onion. 

![lab-net1](https://user-images.githubusercontent.com/47642539/104577244-8813a900-5659-11eb-84eb-f2fca8c95d43.png)

**Comment by defensivedepth** [comment]
I just replayed that pcap on a Standalone system to the `bond0` interface and I get lots of LokiBot Suricata alerts.

I would confirm that you are actually seeing that traffic on `bond0` by running tcpdump on `bond0` and filtering for some known traffic within that pcap, and then try replaying it again.

**Comment by aremai** [comment]
I get the same results as @AlpenFlizzer could you please @defensivedepth explain how you replayed that pcap on SecurityOnion so that you get LokiBot Suricata alerts.

**Reply by defensivedepth** [reply]
Can you share the `tcpreplay` command you are using to replay the pcap?
